web-api/tests/split_documents.rs

```diff
@@ -0,0 +1,448 @@
+                    .build()?,
+                StatusCode::NO_CONTENT,
+                false,
+            )
+            .await;
+
+            let res = query("cars", "filter-a").await?;
+            assert!(res.id_ranking().is_empty());
+
+            send_assert(
+                &client,
+                client
+                    .put(ingestion_url.join("/documents/d1/properties/foo")?)
+                    .json(&json!({
+                        "property": "filter-a",
+                    }))
+                    .build()?,
+                StatusCode::NO_CONTENT,
+                false,
+            )
+            .await;
+
+            let res = query("cars", "filter-a").await?;
+            assert_eq!(res.id_ranking(), vec!["d1"]);
+            assert_eq!(res.documents[0].splits, [0, 1, 2].into_iter().collect());
+
+            send_assert(
+                &client,
+                client
+                    .put(ingestion_url.join("/documents/d2/properties/foo")?)
+                    .json(&json!({
+                        "property": "filter-a",
+                    }))
+                    .build()?,
+                StatusCode::NO_CONTENT,
+                false,
+            )
+            .await;
+
+            let res = query("cars", "filter-a").await?;
+            assert_eq!(res.id_ranking(), vec!["d2", "d1"]);
+            assert_eq!(res.documents[0].splits, [0, 1, 2].into_iter().collect());
+            assert_eq!(res.documents[1].splits, [0, 1, 2].into_iter().collect());
+
+            send_assert(
+                &client,
+                client
+                    .delete(ingestion_url.join("/documents/d1/properties/foo")?)
+                    .build()?,
+                StatusCode::NO_CONTENT,
+                false,
+            )
+            .await;
+
+            let res = query("cars", "filter-a").await?;
+            assert_eq!(res.id_ranking(), vec!["d2"]);
+            assert_eq!(res.documents[0].splits, [0, 1, 2].into_iter().collect());
+
+            Ok(())
+        },
+    );
+}
+
+#[test]
+fn test_endpoints_which_do_not_yet_fully_support_split_do_not_fall_over() {
+    test_two_apps::<Ingestion, Personalization, _>(
+        UNCHANGED_CONFIG,
+        with_dev_options(),
+        |client, ingestion_url, personalization_url, _| async move {
+            ingest(&client, &ingestion_url).await?;
+            send_assert(
+                &client,
+                client
+                    .patch(personalization_url.join("/users/u1/interactions")?)
+                    .json(&json!({ "documents": [ { "id": "d1" } ] }))
+                    .build()?,
+                StatusCode::NO_CONTENT,
+                false,
+            )
+            .await;
+            send_assert_json::<SearchResponse>(
+                &client,
+                client
+                    .post(personalization_url.join("/users/u1/personalized_documents")?)
+                    .json(&json!({}))
```

Comment:
:art: the request body for that endpoint is optional since all keys are optional, hence i implemented the serde of that endpoint such that an empty request body can also be omitted
```suggestion
```